wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/ProcessCommunicationProvider.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.che.api.languageserver.LanguageServerConfig.CommunicationProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation provides communication to a process that runs a language server. Besides that
+ * it also lazily starts the language server process.
+ */
+public class ProcessCommunicationProvider implements CommunicationProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(ProcessCommunicationProvider.class);
+
+  private final ProcessBuilder processBuilder;
+  private final String languageServerName;
+  private final StatusChecker statusChecker;
+
+  private Process process;
```

Comment:
I don't think this variable is thread safe.

Reply:
Me neither.